### ISSUE TITLE
Problum Occurred during parsing CPPcheck output version 1.84

### ISSUE THREAD

**Issue by Hansvdsteen** [kwin/cppcheclipse#90]
The following output syntax is introduced at CPPcheck version 1.84 and will generate a 'Problem Ocurred' pop-up with the reason 'Resource '/project/nofile' does not exist.'

CPPcheck Console output:
nofile;0;information;missingIncludeSystem;Cppcheck cannot find all the include files (use --check-config for details)

The first 2 tokens (nofile and 0) where in previous versions of cppcheclk defined as empty strings. 
It would be nice to support this, so the combination 'nofile' on line '0' is special and is used for general information instead of an error pointing to a real file.

**Comment by truhlikfredy** [comment]
You are saying that you were using cppcheck 1.83 before and that had no problems? And the problem is introduced only with 1.84?

**Comment by Hansvdsteen** [comment]
That's what it looks like yes. Ik could not find anything about it in the release notes of 1.84, buy it seems something had been changed here.

Op 12 sep. 2018 13:12 schreef Anton Krug <notifications@github.com>:

You are saying that you were using cppcheck 1.83 before and that had no problems? And the problem is introduced only with 1.84?

**Comment by Margulis** [comment]
I have the same problem. Cppcheck runs in Eclipse by "cppcheclipse" plugin. Default settings.
Command:
`/usr/bin/cppcheck --template={file};{line};{severity};{id};{message} --enable=all --verbose --platform=unix64 --std=c++11 --file-list=- `

Console output:
`nofile;0;information;missingInclude;Cppcheck cannot find all the include files. Cppcheck can check the code without the include files found. But the results will probably be more accurate if all the include files are found. Please check your project's include directories and add all of them as include directories for Cppcheck. To see what files Cppcheck cannot find use --check-config.
`

**Comment by Hansvdsteen** [comment]
Wondering if someone is looking at/fixing this problem, please do!

**Comment by Margulis** [comment]
The problem is also reproducing in CPPcheck version 1.85.

**Comment by kwin** [comment]
This is open source. I am happy to accept high quality PRs.

**Comment by kwin** [comment]
This should be fixed by https://github.com/kwin/cppcheclipse/commit/0da3574eec3a399e38ed4d09f63330bbb42d4a41.

**Comment by pexus** [comment]
When will this fix be available on the Eclipse Market Place - Is there a binary available somewhere?
Thanks

**Comment by WoutHakvoort** [comment]
Still getting the error which is mentioned above. Cppcheck 1.8.5 and 1.8.6 used. 

![snip](https://user-images.githubusercontent.com/33685691/51780088-fd53df80-210b-11e9-8ba7-9d3a1356c15f.PNG)

![snip 2](https://user-images.githubusercontent.com/33685691/51780092-12c90980-210c-11e9-81d0-e6d9448204fe.PNG)

I have installed cppcheclipse remotely and via Eclipse Marketplace, but both versions show the errors given above.

**Comment by Hansvdsteen** [comment]
Thats because this issue is solved for release 1.1.1, but is not released yet.

**Comment by AntonKrug** [comment]
I'm usually building the artifact myself. But having official build would help as there was none in good while

**Comment by Hansvdsteen** [comment]
@kwin it would be nice to make a release 1.1.1 now.. i have 'a lot' of questions about when this issue will be solved.. And actually for a lot of people it is solved when it is in an official release.

**Comment by AntonKrug** [comment]
I concur, I have to make my own custom build as my PR got accepted to solve one issue I needed to solve, but it's not in the release because latest release 2-3 years old

**Comment by Hansvdsteen** [comment]
If i can help you with something.. please let me know

^Hans van der Steen

**Comment by AntonKrug** [comment]
Thanks, for the offer, I would like to just see a release build to recheck what things work and do not work. On my build i'm using a PR for different issue and for this one with cppcheck 1.84 I just stayed version behind.

**Comment by Hansvdsteen** [comment]
If you say a release build.. you mean 1.1.1?
Is the pull-request on your build needed for 1.1.1? which PR do you actually mean?

^Hans van der Steen

**Comment by AntonKrug** [comment]
Yes the 1.1.1.

>Is the pull-request on your build needed for 1.1.1? 
I do not get this question, can you rephrase it?

It's this PR:
https://github.com/kwin/cppcheclipse/pull/82

**Comment by Hansvdsteen** [comment]
@AntonKrug i mean is your PR something which solves something which may block making a release 1.1.1? My PR (#92) is already taken over.. so actually ready to be released in 1.1.1. @kwin So what is taken so long?

**Comment by AntonKrug** [comment]
No it shouldn't block release of the 1.1.1
It looks like the same situation with your PR

**Comment by AntonKrug** [comment]
@Hansvdsteen I would guess @kwin is very busy, this repo is not his only project and people have life etc...

**Comment by ehoeks** [comment]
If someone can build a BETA for the 1.1.1. I would be glad to test it. Got some big gnu-mcu-eclipse projects that stress the cppcheclipse thoroughly.

**Comment by kwin** [comment]
I am hoping to get around doing a release this weekend.

**Comment by AntonKrug** [comment]
@ehoeks if you have maven installed, you can try to build it yourself. Something like this could work:

`mvn clean package`

**Comment by keyuracharya** [comment]
hi, can anyone please update 1.1.1 to the eclipse market place?  it seems that marketplace is still having 1.1.0 
Unfortunately all those who use the plugin will not have necessary software installed to build at their end.

**Comment by Hansvdsteen** [comment]
@keyuracharya i think 1.1.1 is still not released yet. @kwin do you know when it will be?

**Comment by Hansvdsteen** [comment]
@kwin is there already a plan of making the 1.1.1 release in the near future?

**Comment by kwin** [comment]
@Hansvdsteen Are you interested in getting commit rights of this project and try the release yourself? I currently don't really find time for this...

**Comment by Hansvdsteen** [comment]
@kwin i am not familiar with the release process of eclipse plugins, but if you can give me some info.. i could check it out.

**Comment by truhlikfredy** [comment]
@kwin I think I should be able to do a release for it

**Comment by Hansvdsteen** [comment]
@truhlikfredy if you are able/allowed to release it.. please do..

**Comment by truhlikfredy** [comment]
@Hansvdsteen I'm not allowed to, but was saying to @kwin if he would give me the rights I could.

**Comment by kwin** [comment]
@truhlikfredy I sent you an invite long time ago to your other user name @AntonKrug. I now resend the invite to your new user. You should probably disable the old user then!

**Comment by truhlikfredy** [comment]
I see, I keep switching between them so I missed as I was checking wrong account at the wrong time, I accepted it now, hope that's not too late (could you check if that's accepted?). I will do the 1.1.1 by hand, but I could update the github actions to do automatic release maybe when there is a tag on the master branch? Done it with travis on my own repositories and it works pretty well but the github action should be more stable anyway.

**Comment by kwin** [comment]
Both invitations are still pending: 
![Screenshot 2020-02-26 at 12 18 35](https://user-images.githubusercontent.com/185025/75340165-3ff96580-5892-11ea-89cf-f1b8f8c0a853.png)

Please go again to https://github.com/kwin/cppcheclipse/invitations to accept.

I am fine with automatically creating a release once a tag is there.

**Comment by truhlikfredy** [comment]
Hopefully now they are accepted, thank you very much, tomorrow I will look into the release and then later how to make it automated. I did the automated thing with travis, but not yet with github actions, so I will practice first on a separate repository and when it will be working do it on this repository

**Comment by truhlikfredy** [comment]
So I played with my jfrog account and can upload the release and zipped from your bintry.ant script, only things which doesn't get uploaded is the updates (the composite upload), which depends on getMajorMinorVersion which seems to depend on updates already existing. So my guess is that it might be ok with your repository on bintray. Would you be so kind and add me to the bintray's cppcheclipse organization? My jfrog handle is antonkrugms.

**Comment by AntonKrug** [comment]
Made a 1.1.1 release, hopefully I did not messed up something. Always so worried when working on somebody's else repositories/projects.

**Comment by Hansvdsteen** [comment]
@AntonKrug where can i get/update it from? so i could test/verify it.

**Comment by kwin** [comment]
@AntonKrug Thanks a lot for the efforts, unfortunately I only see version 1.1.0 at the bintray p2 update site as well:
<img width="997" alt="Screenshot 2020-03-03 at 19 46 29" src="https://user-images.githubusercontent.com/185025/75808722-dd1f3700-5d87-11ea-92a7-b169d15738c3.png">

**Comment by AntonKrug** [comment]
I'm very sorry for the hassle. In the meantime adding the zip from here should work https://github.com/kwin/cppcheclipse/releases/download/1.1.1/com.googlecode.cppcheclipse.repository-1.1.1.zip 

And for the update repository I did the:
```
mvn -Dtycho.mode=maven org.eclipse.tycho:tycho-versions-plugin:set-version -DnewVersion=1.1.1
mvn clean verify -Prelease-composite
```

I see it in the zip/release folders in the bintray (however without the timestamp postfix), and the composite (for the update folder) run (which I couldn't before because the getMajorMinorVersion was not able to fetch):
```
getMajorMinorVersion:

push-composite-to-bintray:
    [apply]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    [apply]                                  Dload  Upload   Total   Spent    Left  Speed
    [apply] 
    [apply]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   514    0     0  100   514      0    328  0:00:01  0:00:01 --:--:--   327
    [apply] {"message":"success"}100   535  100    21  100   514     12    311  0:00:01  0:00:01 --:--:--   324
    [apply]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    [apply]                                  Dload  Upload   Total   Spent    Left  Speed
    [apply] 
    [apply]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   514    0     0  100   514      0    344  0:00:01  0:00:01 --:--:--   344
    [apply] {"message":"success"}100   535  100    21  100   514     11    290  0:00:01  0:00:01 --:--:--   302

push-main-composite-to-bintray:
    [apply]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    [apply]                                  Dload  Upload   Total   Spent    Left  Speed
    [apply] 
    [apply]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0100   491    0     0  100   491      0    308  0:00:01  0:00:01 --:--:--   308
    [apply] {"message":"success"}100   512  100    21  100   491     11    278  0:00:01  0:00:01 --:--:--   289100   512  100    21  100   491     11    278  0:00:01  0:00:01 --:--:--   289
    [apply]   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
    [apply]                                  Dload  Upload   Total   Spent    Left  Speed
    [apply] 
    [apply]   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0  0   491    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
    [apply] {"message":"success"}100   512  100    21  100   491     12    283  0:00:01  0:00:01 --:--:--   295100   512  100    21  100   491     12    283  0:00:01  0:00:01 --:--:--   295
[INFO] Executed tasks
```

**Comment by AntonKrug** [comment]
With the child and main composites I was struggling the most, will have check the getMajorMinorVersion javascript if it still works and I struggle with the timestamp as well. Not sure if some features of the maven changed over the time. However in the mean time I updated the XMLs by hand and now the update get recognized and it does install if you do not have 1.1.1 already installed
![1 1 1](https://user-images.githubusercontent.com/30338596/75833837-2ab09980-5db2-11ea-9045-87cc6bc71dca.png)

**Comment by Hansvdsteen** [comment]
@AntonKrug i was able to download 1.1.1 now! Thanks for the hard work!.
Only thing which still shows 1.1.0 version is in the market place. Don't know how this exactly works.
![image](https://user-images.githubusercontent.com/41508065/75854984-96febd80-5df1-11ea-9ce3-13f1a2997a3e.png)

**Comment by AntonKrug** [comment]
I can update the wiki, at least some draft and let @kwin approve the final touches. But will have to look up how to do the marketplace. And still not happy that my release is missing timestamp.

**Comment by kwin** [comment]
I updated the marketplace entry. @AntonKrug feel free to send me your eclipse.org account, then I can make you editor of the marketplace entry as well.

**Comment by truhlikfredy** [comment]
My marketplace handle is akrug. Will be ok with you for me to update the Wiki as well? And I specified java8 for the travis build, so now it is again working, I would say the github actions you added is the future and will be the one with the least hassle to keep going, but shouldn't hurt to have 2 different system just as extra sanity check. What is your preference?

**Comment by kwin** [comment]
@truhlikfredy Please merge your different Github accounts (https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/merging-multiple-user-accounts) it is really confusing to use two different accounts.
I added `akrug` as editor to the marketplace entry, you can now edit it at https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/merging-multiple-user-accounts.

Feel free to update the wiki as well (preferably via PR). Regarding CI I would also rely on GH Actions in the future. Maybe we can also set up some automated SNAPSHOT deployment to bintray.
